### DISCUSSION TITLE
Cannot include/using from within a namespace or class

### DISCUSSION THREAD

**Discussion by elaine-jackson** [nlohmann/json#2381] [Q&A] [answered]
When using json.hpp (single include) if I include it from inside a namespace or a class like the following example. I'm unsure if this is intended behavior or a bug.

```
namespace Modals {
  #include <../lib/json.hpp>
  using json = nlohmann::json;
  class Person {
    ...
  }
}
```

My Makefile is... using the C++17 standard.
```
CXX       := g++
CXX_FLAGS := -Wall -Wextra -std=c++17 -ggdb

BIN     := bin
SRC     := src
INCLUDE := include
LIB     := lib
LIBRARIES   := 
EXECUTABLE  := main

all: $(BIN)/$(EXECUTABLE)

run: clean all
	clear
	@echo "🚀 Executing..."
	./$(BIN)/$(EXECUTABLE)

$(BIN)/$(EXECUTABLE): $(SRC)/*.cpp
	@echo "🚧 Building..."
	$(CXX) $(CXX_FLAGS) -I$(INCLUDE) -L$(LIB) $^ -o $@ $(LIBRARIES) -lcurl

clean:
	@echo "🧹 Clearing..."
	-rm $(BIN)/*
```

I've included the error log for further analysis.
[errors.txt](https://github.com/nlohmann/json/files/5164777/errors.txt)

**Comment by gregmarr** [comment, marked as answer]
You definitely shouldn't do that.  You've just introduced all of these types inside the `Modals` namespace when they expect to be in the global namespace.  This extends not just to the types from `json.hpp` but also to all the types in all the standard headers included by the `json.hpp` header.
